packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -278,9 +292,14 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
       mounted.current = true;
       return;
     }
-    const value = options[selectedIndex]?.value;
-    onChangeProp.current?.(value);
-  }, [options, selectedIndex]);
+    const currentValue = optionsRef.current[selectedIndex]?.value;
+
+    // If current value is the same as selected value, do not call user defined onChange
```

Comment:
nit: This comment is redundant as there is a similar comment right above the `useEffect`.

Reply:
Removed